core/control-plane/contract-core/src/test/java/org/eclipse/edc/connector/contract/offer/ContractOfferResolverImplIntegrationTest.java

```diff
@@ -107,6 +107,35 @@ void shouldLimitResult_withHeterogenousChunks() {
         verify(policyStore).findById("contract");
     }
 
+    @Test
+    void shouldReturnOfferSubsetAcrossMultipleContractDefinitions() {
+
+        var assets1 = range(1, 11).mapToObj(i -> createAsset("asset-" + i).build()).collect(Collectors.toList());
+        var assets2 = range(11, 21).mapToObj(i -> createAsset("asset-" + i).build()).collect(Collectors.toList());
+
+        store(assets1);
+        store(assets2);
+
+        var contractDefinition1 = getContractDefBuilder("contract-definition-")
+                .selectorExpression(selectorFrom(assets1)).build();
+        var contractDefinition2 = getContractDefBuilder("contract-definition-")
+                .selectorExpression(selectorFrom(assets2)).build();
+
+        when(agentService.createFor(isA(ClaimToken.class))).thenReturn(new ParticipantAgent(emptyMap(), emptyMap()));
+        when(contractDefinitionService.definitionsFor(isA(ParticipantAgent.class))).thenAnswer(i -> Stream.of(contractDefinition1, contractDefinition2));
+        when(policyStore.findById(any())).thenReturn(PolicyDefinition.Builder.newInstance().policy(Policy.Builder.newInstance().build()).build());
+
```

Comment:
those two assertions seem to test the same thing, i.e. getting contract offers that "span" across assets that are referred to by two contract definitions. Therefor one of them could be removed. However, we should actually test edge cases too:
- `0...12`
- `0...9999`: tests a range overshoot, expect 20 offers
- `4...9999`: test a range overshoot starting at an offset, expect 16 offers
- `3...20`: test exact upper bound with offset, expect 17 offers

At that point you may want to consider a `@ParameterizedTest` to avoid test code duplication.

Reply:
thanks, @paullatzelsperger.. doing it.

Reply:
we're getting there :) check updated comments.